packages/world/src/utils/PlayerSkillUtils.sol

```diff
@@ -0,0 +1,164 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.24;
+
+import { FixedPointMathLib } from "solady/utils/FixedPointMathLib.sol";
+
+import {
+  SKILL_BUILD_ENERGY_TO_MAX,
+  SKILL_CRAFT_MASS_ENERGY_TO_MAX,
+  SKILL_ENERGY_MAX_DISCOUNT_WAD,
+  SKILL_FALL_ENERGY_TO_MAX,
+  SKILL_HIT_MACHINE_ENERGY_TO_MAX,
+  SKILL_HIT_PLAYER_ENERGY_TO_MAX,
+  SKILL_MINING_MASS_TO_MAX,
+  SKILL_MOVE_ENERGY_TO_MAX
+} from "../Constants.sol";
+import { ActivityType } from "../codegen/common.sol";
+
+import { EntityId } from "../types/EntityId.sol";
+import { ObjectType, ObjectTypes } from "../types/ObjectType.sol";
+import { PlayerProgressUtils as Tracking } from "./PlayerProgressUtils.sol";
+
+library PlayerSkillUtils {
+  function getMoveEnergyMultiplierWad(EntityId player) internal view returns (uint256) {
+    return getEnergyMultiplierWad({
+      progress: Tracking.getProgress(player, ActivityType.MoveEnergy),
+      progressCap: SKILL_MOVE_ENERGY_TO_MAX
+    });
+  }
+
+  function getFallEnergyMultiplierWad(EntityId player) internal view returns (uint256) {
+    return getEnergyMultiplierWad({
+      progress: Tracking.getProgress(player, ActivityType.MoveFallEnergy),
+      progressCap: SKILL_FALL_ENERGY_TO_MAX
+    });
+  }
+
+  function getMineEnergyMultiplierWad(EntityId player, ObjectType toolType, ObjectType minedType)
+    internal
+    view
+    returns (uint256)
+  {
+    ActivityType activityType;
+    if (minedType.isCrop()) {
+      activityType = ActivityType.MineCropMass;
+    } else if (toolType.isAxe()) {
+      activityType = ActivityType.MineAxeMass;
+    } else if (toolType.isPick()) {
+      activityType = ActivityType.MinePickMass;
+    } else {
+      // TODO: use wad
```

Comment:
can we just import Wad from solady?

Reply:
yup I actually wrapped most of solady stuff in our Math lib now, so we don't have to mix both. Also adding WAD to our constants